src/map/ai/controllers/trust_controller.cpp

```diff
@@ -127,4 +146,48 @@ bool CTrustController::Ability(uint16 targid, uint16 abilityid)
         return POwner->PAI->Internal_Ability(targid, abilityid);
     }
     return false;
-}
\ No newline at end of file
+}
+
+CBattleEntity* CTrustController::GetTopEnmity()
+{
+    CBattleEntity* PEntity = nullptr;
+    if (auto PMob = dynamic_cast<CMobEntity*>(POwner->PMaster->GetBattleTarget()))
+    {
+        return PMob->PEnmityContainer->GetHighestEnmity();
+    }
+    return PEntity;
+}
+
+position_t CTrustController::GetDeclumpedPosition(position_t target_pos)
+{
+    auto lerp = [](float a, float b, float f)
+    {
+        return a + f * (b - a);
+    };
+
+    auto lerp_position = [&](position_t a, position_t b, float ratio)
+    {
+        float safe_ratio = std::clamp(ratio, 0.0f, 1.0f);
+        position_t final_pos{
+            lerp(a.x, b.x, ratio),
+            lerp(a.y, b.y, ratio),
+            lerp(a.z, b.z, ratio),
+            0, 0};
+        return final_pos;
+    };
+
+    position_t final_pos = target_pos;
+    for (auto PTrust : static_cast<CCharEntity*>(POwner->PMaster)->PTrusts)
```

Comment:
Like my comment above, there are still instances of trust dance parties and them getting stuck in interesting situations... 

This will be a pretty hefty TODO in the future, but it's fun to play with right now